src/inspector_agent.h

```diff
@@ -23,6 +23,21 @@ class Environment;
 
 namespace inspector {
 
+class ChannelImpl;
+class InspectorIo;
+class NodeInspectorClient;
+
+class InspectorSession {
```

Comment:
Please add some documentation to make this more maintainable. Namely, I'd like to see

1. What this class is for? Especially how it differs from the similarly named DispatcherSession.
2. Who creates this class?
3. Who will own this class?
4. Who will hold references to this class, other than its owner?

As I understand this class

1. is for the session delegate (WebSocket or JS binding) to control a connection to the inspector Agent, both for sending messages through `Dispatch()` and to disconnect from the inspector Agent by `delete`ing an object of this class,
2. is created when the Agent connects to a session delegate
3. is owned by either the session delegate itself or the owner of the session delegate
4. other classes do not have references to it

if this is correct then please articulate it. And please do so for all the newly created classes in this PR.